src/snowflake/snowpark/_internal/data_source/datasource_partitioner.py

```diff
@@ -0,0 +1,246 @@
+        self.dialect = DBMS_MAP[dbms_type]()
+        self.driver = DRIVER_MAP[driver](create_connection)
+        self.dialect_class = DBMS_MAP[dbms_type]
+        self.driver_class = DRIVER_MAP[driver]
+
+    def reader(self) -> DataSourceReader:
+        return DataSourceReader(
+            self.driver_class, self.create_connection, self.schema, self.fetch_size
+        )
+
+    @property
+    def schema(self) -> StructType:
+        if self.custom_schema is None:
+            return self.driver.infer_schema_from_description(self.table_or_query)
+        else:
+            if isinstance(self.custom_schema, str):
+                schema = type_string_to_type_object(self.custom_schema)
+                if not isinstance(schema, StructType):
+                    raise ValueError(
+                        f"Invalid schema string: {self.custom_schema}. "
+                        f"You should provide a valid schema string representing a struct type."
+                        'For example: "id INTEGER, int_col INTEGER, text_col STRING".'
+                    )
+                return schema
+            elif isinstance(self.custom_schema, StructType):
+                return self.custom_schema
+            else:
+                raise ValueError(
+                    f"Invalid schema type: {type(self.custom_schema)}."
+                    'The schema should be either a valid schema string, for example: "id INTEGER, int_col INTEGER, text_col STRING".'
+                    'or a valid StructType, for example: StructType([StructField("ID", IntegerType(), False)])'
+                )
+
+    @property
+    def partitions(self) -> List[str]:
+        select_query = self.dialect.generate_select_query(
+            self.table_or_query, self.schema
+        )
+        logger.debug(f"Generated select query: {select_query}")
+        if self.column is None:
+            if (
+                self.lower_bound is not None
+                or self.upper_bound is not None
+                or self.num_partitions is not None
+            ):
+                raise ValueError(
+                    "when column is not specified, lower_bound, upper_bound, num_partitions are expected to be None"
+                )
+            if self.predicates is None:
+                partitioned_queries = [select_query]
+            else:
+                partitioned_queries = self.generate_partition_with_predicates(
+                    select_query, self.predicates
+                )
+        else:
+            if (
+                self.lower_bound is None
+                or self.upper_bound is None
+                or self.num_partitions is None
+            ):
+                raise ValueError(
+                    "when column is specified, lower_bound, upper_bound, num_partitions must be specified"
+                )
+
+            column_type = None
+            for field in self.schema.fields:
+                if field.name.lower() == self.column.lower():
```

Comment:
this could be a bug if the source db supports case sensitive identifiers, `col_A` vs `col_a` are two different columns but are treated as identical in this part.
 let's log a jira for this

Reply:
I updated this part to convert input column name into snowflake style than make the comparision